erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -100,16 +100,13 @@ def process_route(self, optimize):
 			optimize (bool): True if route needs to be optimized, else False
 		"""
 
-		if not frappe.db.get_single_value("Google Maps Settings", "enabled"):
-			frappe.throw(_("Cannot process route, since Google Maps Settings is disabled."))
-
 		departure_datetime = get_datetime(self.departure_time)
 		route_list = self.form_route_list(optimize)
 
 		# For locks, maintain idx count while looping through route list
 		idx = 0
 		for route in route_list:
-			directions = get_directions(route, optimize)
+			directions = get_directions(self, route, optimize)
```

Comment:
If the document is required, convert the `get_directions` function into a method in the class.